hoomd/write/gsd_burst.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2009-2023 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Write GSD last :math:`N` frames at user direction."""
+
+from hoomd import _hoomd
+from hoomd.filter import All
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.write.gsd import GSD
+
+
+class Burst(GSD):
+    r"""Write last :math:`N` stored frames at user trigger in the GSD format.
+
+    This class stores the last :math:`N` frames in an interal deque
+    which only writes the frames out when `dump` is called. When the writer is
+    triggered and the next frame would result in :math:`N + 1` frames being
+    stored, the oldest frame is popped off the deque and the new frame is
+    pushed to the front of the deque.
+
+    Args:
+        trigger (hoomd.trigger.trigger_like): Select the timesteps to store
+            configuration.
+        filename (str): File name to write to when calling `dump`.
+        filter (hoomd.filter.filter_like): Select the particles to write.
+            Defaults to `hoomd.filter.All`.
+        mode (str): The file open mode. Defaults to ``'ab'``.
+        dynamic (list[str]): Field names and/or field categores to save in
+            all frames. Defaults to ``['property']``.
+        logger (hoomd.logging.Logger): Provide log quantities to write. Defaults
+            to `None`.
+        max_burst_frames (int): The maximum number of frames to store before
+            between writes. -1 represents no limit. Defaults to -1.
+
```

Comment:
Add a note that when there are no frames in the GSD file, `Burst` always one frame the first time it is triggered to capture all the static quantities.